### PR TITLE
Prioritization edits

### LINKED ISSUE #2321: "PRIORITY frame expresses priority of a stream"
Draft 17 says

> The PRIORITY (type=0x02) frame specifies the client-advised priority of a stream.

If I understand H3 Placeholders correctly, the PRIORITY frame would be used to set their priority too. Perhaps we can say:

`The PRIORITY (type=0x02) frame specifies the client-advised priority of a stream or placeholder`.

### PR DESCRIPTION
**Pull request by LPardue**
Fixes #2321 

Maybe takes things too far, please discuss.

### CODE REVIEW

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -1041,41 +1044,47 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
+of the tree. A reference to an element which is no longer in the tree is treated
+as a reference to the root of the tree. The structure of the dependency tree
+changes as PRIORITY frames modify the dependency links between prioritized
+elements.
 
 Due to reordering between streams, an element can also be prioritized which is
 not yet in the tree. Such elements are added to the tree with the requested
 priority.
 
+
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]: Extra blank line here.

File: draft-ietf-quic-http.md [hunk 2]
```diff
@@ -1041,41 +1044,47 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
+In all cases, elements are assigned an initial weight of 16 unless a PRIORITY
+frame begins the stream.
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]: That's only possible for request streams.

File: draft-ietf-quic-http.md [hunk 3]
```diff
@@ -1041,41 +1044,47 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
+
+When a client request is first sent, its parent and weight are determined by the
+PRIORITY frame (see {{frame-priority}}) which begins the stream, if present.
+Otherwise, the element is dependent on the root of the priority tree. A request 
+stream dependent upon another request stream expresses the preference that the
+latter stream (the "parent" request) be allocated resources before the former
+stream (the "dependent" request).
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]: This paragraph feels like the thing that needs improvement in this section.

File: draft-ietf-quic-http.md (hunk 2 above)
Comment (LPardue) [reply draft-ietf-quic-http.md]: That's what the old text said, I just moved it. 

Should I adjust the text somehow to make things clearer?

File: draft-ietf-quic-http.md (hunk 3 above)
Comment (LPardue) [reply draft-ietf-quic-http.md]: This is almost a worked example that helps people familiar with H2's priority scheme

File: draft-ietf-quic-http.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-quic-http.md]:
```suggestion
```

File: draft-ietf-quic-http.md [hunk 4]
```diff
@@ -1025,41 +1029,39 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
+changes as PRIORITY frames modify the dependency links between prioritized
+elements.
 
 Due to reordering between streams, an element can also be prioritized which is
 not yet in the tree. Such elements are added to the tree with the requested
 priority.
 
+When a prioritized element is first created it has a default inital weight
+of 16 and a default dependency. Requests and placeholders are dependent on the
+root of the priority tree, pushes are dependent on the client request on which
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]:
```suggestion
root of the priority tree; pushes are dependent on the client request on which
```
Comma splice

File: draft-ietf-quic-http.md (hunk 1 above)
Comment (MikeBishop) [reply draft-ietf-quic-http.md]: The extra line was removed in your most recent commit, so don't accept the suggestion to remove another one.  ;-)

File: draft-ietf-quic-http.md [hunk 5]
```diff
@@ -1025,41 +1029,39 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
+the PUSH_PROMISE frame was sent.
+
+Requests may override the default intial values by including a PRIORTIY frame
+(see {{frame-priority}}) at the beginning of the stream. Pushes and placeholders
+may be later updated by sending a PRIORITY frame on the control stream.
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]:
```suggestion
can be updated by sending a PRIORITY frame on the control stream.
```

File: draft-ietf-quic-http.md [hunk 6]
```diff
@@ -1025,41 +1029,39 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
 not yet in the tree. Such elements are added to the tree with the requested
 priority.
 
+When a prioritized element is first created it has a default inital weight
+of 16 and a default dependency. Requests and placeholders are dependent on the
+root of the priority tree, pushes are dependent on the client request on which
+the PUSH_PROMISE frame was sent.
+
+Requests may override the default intial values by including a PRIORTIY frame
+(see {{frame-priority}}) at the beginning of the stream. Pushes and placeholders
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]:
```suggestion
(see {{frame-priority}}) at the beginning of the stream. These priorities
```
It's not just pushes and placeholders -- this statement is true for any of them.

File: draft-ietf-quic-http.md [hunk 7]
```diff
@@ -1025,41 +1029,39 @@ RST bit set, as a stream error of type HTTP_CONNECT_ERROR
...
+of the tree. A reference to an element which is no longer in the tree is treated
+as a reference to the root of the tree. The structure of the dependency tree
+changes as PRIORITY frames modify the dependency links between prioritized
+elements.
 
 Due to reordering between streams, an element can also be prioritized which is
 not yet in the tree. Such elements are added to the tree with the requested
 priority.
 
+When a prioritized element is first created it has a default inital weight
```
Comment (MikeBishop) [inline draft-ietf-quic-http.md]:
```suggestion
When a prioritized element is first created, it has a default initial weight
```

**Pull request merged**